cookbooks/anisotropic_viscosity/av_material.cc

```diff
@@ -0,0 +1,858 @@
+                    material_model_outputs.viscosities[q] / disl_viscosities_out->dislocation_viscosities[q];
+                }
+                
+                heating_model_outputs.lhs_latent_heat_terms[q] = 0.0;
+            }
+        }
+        
+        template <int dim>
+        void
+        ShearHeatingAnisotropicViscosity<dim>::
+        create_additional_material_model_outputs(MaterialModel::MaterialModelOutputs<dim> &material_model_outputs) const
+        {
+            const unsigned int n_points = material_model_outputs.viscosities.size();
+            
+            if (material_model_outputs.template get_additional_output<MaterialModel::AnisotropicViscosity<dim> >() == nullptr)
+            {
+                material_model_outputs.additional_outputs.push_back(
+                                                                    std::make_shared<MaterialModel::AnisotropicViscosity<dim>> (n_points));
+            }
+            
+            this->get_material_model().create_additional_named_outputs(material_model_outputs);
+        }
+    }
+    
+    namespace MaterialModel
+    {
+        using namespace dealii;
+        
+        template <int dim>
+        class Anisotropic : public MaterialModel::Simple<dim>
+        {
+        public:
+            virtual void initialize();
+            
+            virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                                  MaterialModel::MaterialModelOutputs<dim> &out) const;
+            static void declare_parameters(ParameterHandler &prm);
+            virtual void parse_parameters(ParameterHandler &prm);
+            
+            /**
+             * Return true if the compressibility() function returns something that
+             * is not zero.
+             */
+            virtual bool
+            is_compressible () const;
+            
+        private:
+            void set_assemblers(const SimulatorAccess<dim> &,
+                                Assemblers::Manager<dim> &assemblers) const;
+            
+            // Constitutive tensor
+            SymmetricTensor<4,dim> C;
+        };
+        template <int dim>
+        class AV : public MaterialModel::Simple<dim>
+            {
+              public:
+            	virtual void initialize();
+                virtual void evaluate (const MaterialModel::MaterialModelInputs<dim> &in,
+                                       MaterialModel::MaterialModelOutputs<dim> &out) const;
+                static void declare_parameters (ParameterHandler &prm);
+                virtual void parse_parameters (ParameterHandler &prm);
+                virtual bool is_compressible () const;
+                virtual double reference_viscosity () const;
+                virtual double reference_density () const;
+                virtual void create_additional_named_outputs(MaterialModel::MaterialModelOutputs<dim> &out) const;
+              private:
+                double eta_N, eta_0;

```

Comment:
can you document what these variables are? I think that your `eta_0` is the ratio of viscosities, but then `eta` is maybe not such a good choice for the symbol/variable name. I would call it `viscosity_ratio`.